src/relay/ir/expr.cc

```diff
@@ -330,6 +330,31 @@ TVM_STATIC_IR_FUNCTOR_REGISTER(IRPrinter, vtable)
   p->stream << "RefWriteNode(" << node->ref << ", " << node->value << ")";
 });
 
+
+Fatal FatalNode::make(std::string msg, Type type_annotation) {
+  NodePtr<FatalNode> n = make_node<FatalNode>();
+  n->msg = std::move(msg);
+  n->type_annotation = std::move(type_annotation);
+  return Fatal(n);
+}
+
+TVM_REGISTER_NODE_TYPE(FatalNode);
+
+TVM_REGISTER_API("relay._make.Fatal")
+.set_body_typed(FatalNode::make);
+
+std::string NoMatchMsg() {
```

Comment:
I feel we can just use constexpr or an enum to define some common error messages in both C++ and python. Not necessary to define such an API.
otherwise lgtm

Reply:
I have to copy the string twice if it is a constant though. I feel like api is cleaner and not cumbersome to use too.